lib/rules/number-max-precision/__tests__/index.js

```diff
@@ -41,7 +41,7 @@ testRule({
 			code: "@IMPORT '1.123.css'",
 		},
 		{
-			code: 'a { background: url(1.123.jpg) }',
+			code: 'a { background: url(1.123.jpg) url("foo" 1.123.jpg) }',
 		},
```

Comment:
suggestion

maybe add `image-set`?
https://developer.mozilla.org/en-US/docs/Web/CSS/image/image-set

Reply:
The tests for `url` are a bit iffy in general.
I think these were intended to catch an implementation detail in `postcss-value-parser`. Not because anyone really wants to ignore floating point precision in numbers in `url`.

In `postcss-value-parser` `url(...)` is always parsed as a function and it's contents can appear to be numbers. e.g. `url(123.jpg)`

This however is incorrect and the CSS specification is very different.

Either `url(...)` is a single `url` token and it's value is a url or it is a function with a name of `url` and it contains a value list that starts with a string token.

- `url(123.jpg)` -> url token
- `url("123.jpg")` -> function with a name of url

Because `@csstools/css-tokenizer` and `@csstools/css-parser-algorithm` follow the CSS specification they will never produce a `url` function where the url value is parsed as a number.

-------

This is also why I had to add a [url modified](https://www.w3.org/TR/css-values-4/#url-modifiers) `url("foo" 1.123.jpg);`. Without this the test coverage wasn't 100%.

Currently no url modifiers exist in the CSS specification, so this isn't testing realistic CSS anymore.

-------

TL;DR;

Either we should remove the special handling of `url` or we should leave it as is.
I don't think we should add more tests for something that is actually an implementation detail in a parser that (as of this change) isn't used anymore :)

Reply:
suggestion

Add a concise one sentence _why_ comment above the `if` that ignores `url`?

Reply:
I've added a note, please let me know if this is unclear in any way.